server/src-lib/Hasura/GraphQL/RemoteServer.hs

```diff
@@ -159,43 +158,10 @@ mergeQueryRoot :: GS.GCtx -> GS.GCtx -> VT.ObjTyInfo
 mergeQueryRoot a b = GS._gQueryRoot a <> GS._gQueryRoot b
 
 mergeMutRoot :: GS.GCtx -> GS.GCtx -> Maybe VT.ObjTyInfo
-mergeMutRoot a b =
-  let objA' = fromMaybe mempty $ GS._gMutRoot a
-      objB  = fromMaybe mempty $ GS._gMutRoot b
-      objA  = newRootOrEmpty objA' objB
-      merged = objA <> objB
-  in bool (Just merged) Nothing $ merged == mempty
-  where
-    newRootOrEmpty x y =
-      if x == mempty && y /= mempty
-      then mkNewEmptyMutRoot
-      else x
-
-mkNewEmptyMutRoot :: VT.ObjTyInfo
-mkNewEmptyMutRoot = VT.ObjTyInfo (Just "mutation root")
-                    (G.NamedType "mutation_root") Set.empty Map.empty
```

Comment:
`Maybe a` has [Semigroup instance](http://hackage.haskell.org/package/base-4.12.0.0/docs/src/GHC.Base.html#line-403). So, `<>` can be directly used to merge `Maybe mutationRoot`s here. No need for `mkNewEmptyMutRoot`.